File: robocode-tankroyale-server/src/main/java/dev/robocode/tankroyale/server/mappers/BotHandshakeToBotInfoMapper.java
package dev.robocode.tankroyale.server.mappers;

import java.util.Collections;

import dev.robocode.tankroyale.schema.BotHandshake;
import dev.robocode.tankroyale.schema.BotInfo;

public final class BotHandshakeToBotInfoMapper {

	private BotHandshakeToBotInfoMapper() {}

	public static BotInfo map(BotHandshake botHandshake, String hostName, Integer port) {
		BotInfo botInfo = new BotInfo();

		botInfo.setName(botHandshake.getName());
		botInfo.setVersion(botHandshake.getVersion());
		botInfo.setAuthor(botHandshake.getAuthor());
		botInfo.setDescription(botHandshake.getDescription());
		botInfo.setUrl(botHandshake.getUrl());
		botInfo.setCountryCode(botHandshake.getCountryCode());
		botInfo.setGameTypes(Collections.unmodifiableList(botHandshake.getGameTypes()));
		botInfo.setPlatform(botHandshake.getPlatform());
		botInfo.setProgrammingLang(botHandshake.getProgrammingLang());
		botInfo.setHost(hostName);
		botInfo.setPort(port);

		return botInfo;
	}
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/mapper/BotHandshakeToBotInfoMapper.kt
package dev.robocode.tankroyale.server.mapper

import dev.robocode.tankroyale.schema.BotHandshake
import dev.robocode.tankroyale.schema.BotInfo
import java.util.*

object BotHandshakeToBotInfoMapper {
    fun map(botHandshake: BotHandshake, hostName: String, port: Int): BotInfo {
        val botInfo = BotInfo()
        botHandshake.apply {
            botInfo.name = name
            botInfo.version = version
            botInfo.author = author
            botInfo.description = description
            botInfo.url = url
            botInfo.countryCode = countryCode
            botInfo.gameTypes = Collections.unmodifiableList(gameTypes)
            botInfo.platform = platform
            botInfo.programmingLang = programmingLang
            botInfo.host = hostName
            botInfo.port = port
        }
        return botInfo
    }
}